strawberryfields/gbs/plot.py

```diff
@@ -343,3 +344,76 @@ def points(
         f = go.Figure(data=[p], layout=layout)
 
     return f
+
+
+def spectrum(
+    energies: list, gamma: float = 100.0, xmin: float = None, xmax: float = None
+):  # pragma: no cover
+    """Plots a spectrum based on input sampled energies.
+
+    Args:
+        energies (list[float]): a list of sampled energies
+        gamma (float): parameter specifying the width of the Lorentzian function
+        xmin (float): minimum limit of the x axis
+        xmax (float): maximum limit of the x axis
+
+    Returns:
+         Figure: spectrum in the form of a histogram of energies with a Lorentzian-function curve
+    """
+    try:
+        import plotly.graph_objects as go
+    except ImportError:
+        raise ImportError(plotly_error)
+
+    if xmin is None:
+        xmin = min(energies) - 1000.0
```

Comment:
It's probably safest to change the x limits as a function of the range of energies. Something like:
`emin = min(energies)`
`emax = max(energies)`
`xmin = emin - 0.05*(emax-emin)`
`xmax = emax + 0.05*(emax-emin)`
The actual number 0.05 (5% of the width) should be chosen carefully. Maybe 10% or 1% is better